src/main/java/com/worktracker/service/impl/TaskServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.worktracker.service.impl;
+
+import com.worktracker.exception.WorktrackerException;
+import com.worktracker.model.Project;
+import com.worktracker.model.Task;
+import com.worktracker.model.dto.TaskRequestDTO;
+import com.worktracker.repository.TaskRepository;
+import com.worktracker.service.ProjectService;
+import com.worktracker.service.TaskService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TaskServiceImpl implements TaskService {
+
+    private final TaskRepository taskRepository;
+    private final ProjectService projectService;
+
+
+    @Autowired
+    public TaskServiceImpl(TaskRepository taskRepository, ProjectService projectService) {
+        this.taskRepository = taskRepository;
+        this.projectService = projectService;
+    }
+
+    @Override
+    public Task createTask(TaskRequestDTO taskRequestDTO) {
+        Project project = projectService.getProject(taskRequestDTO.getProjectID());
+        if (project == null) {
```

Comment:
After the new update on the method `getProject` it will never return null. So this check is redundant.